Remove manual symlink/clone requirement

ryangoree opened the pull request:
Related to https://github.com/delvtech/hyperdrive/issues/978

# Description

This PR removes the requirement to manually symlink or clone the hyperdrive repo by modifying the `hyperdrive_wrappers` build script to clone it if it doesn't already exist using `HYPERDRIVE_REPO_URL` and `HYPERDRIVE_REF` environment variables.

dpaiton commented on crates/hyperdrive-wrappers/build.rs:
```diff
@@ -56,40 +58,41 @@ const TARGETS: &[&str] = &[
+    // Re-run this script whenever .env, the build script itself, or a contract changes.
+    println!("cargo:rerun-if-changed=.env");
     println!("cargo:rerun-if-changed=build.rs");
     println!("cargo:rerun-if-changed=hyperdrive/contracts/");
```
`println!("cargo:rerun-if-changed=hyperdrive/contracts/");`
is this going to work if the repo is not cloned beforehand?

dpaiton replied:
I think we can probably just remove this line

dpaiton commented on .github/workflows/rust_test.yml:
```diff
@@ -38,16 +38,6 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: check out hyperdrive
```
if this works then you can remove the code above
```
      - name: Fetch latest release tag from Hyperdrive
        run: ./scripts/fetch-latest-tag.sh delvtech hyperdrive
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

ryangoree replied:
This is still needed to set the `HYPERDRIVE_REF` environment variable to the latest release below.

dpaiton commented on .github/workflows/rust_lint.yml:
```diff
@@ -38,16 +38,6 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: check out hyperdrive
```
And I guess remove the script, although I was proud of that script :(

ryangoree replied:
We still need it actually! See earlier comment. And it *is* a cool script 💪

dpaiton commented on crates/hyperdrive-wrappers/.env.example:
```diff
@@ -0,0 +1,2 @@
+HYPERDRIVE_REPO_URL=https://github.com/delvtech/hyperdrive.git
```
is there going to be an install step to move this to `.env` or do we read directly from `.env.example`? If the latter, why not just call it `.env.`?

dpaiton reviewed (approved):
awesome, much cleaner!

You probably want to update the readme. We still need devs to symlink, so maybe readme is fine. But it could be worth noting that CI handles this by pulling the env version directly.

ryangoree commented:
> awesome, much cleaner!
> 
> You probably want to update the readme. We still need devs to symlink, so maybe readme is fine. But it could be worth noting that CI handles this by pulling the env version directly.

🙌 

I updated the README to include instructions on setting up your environment and moved the symlinking instructions to a callout (quote) about local development.
